### ISSUE TITLE
[Bug][Gitlab] Error during fetching from GitLab 

### ISSUE THREAD

**Issue by antoniomuso** [apache/devlake#7194]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Fetching from Gitlab sometimes crashes with the following error:
```
time="2024-03-17 00:02:45" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] get data from _raw_gitlab_api_merge_request_notes where params={\"ConnectionId\":1,\"ProjectId\":1328} and got 3112"
time="2024-03-17 00:02:51" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] finished records: 1"
time="2024-03-17 00:02:58" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] finished records: 500"
time="2024-03-17 00:03:00" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] finished records: 1000"
time="2024-03-17 00:03:03" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] finished records: 1500"
time="2024-03-17 00:03:03" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] finished records: 2000"
time="2024-03-17 00:03:06" level=info msg=" [pipeline service] [pipeline #111] [task #634] [extractApiMergeRequestsNotes] finished records: 3000"
time="2024-03-17 00:03:09" level=error msg=" [pipeline service] [pipeline #111] [task #634] subtask extractApiMergeRequestsNotes ended unexpectedly\n\tWraps: (2) Error 1213 (40001): Deadlock found when trying to get lock; try restarting transaction (500)\n\tWraps: (3) Error 1213 (40001): Deadlock found when trying to get lock; try restarting transaction\n\tError types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *mysql.MySQLError"
```

### What do you expect to happen

It should not crash.

### How to reproduce

I don't know why it happens. Currently, I have this error only for one project.

### Version

v0.20.0-beta10

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by klesh** [comment]
Interesting, were there any other pipelines running in parallel?
Maybe related to #7099

**Comment by antoniomuso** [comment]
> Interesting, were there any other pipelines running in parallel? Maybe related to #7099

Yes, I think so because this happens during the schedule. I currently have two projects scheduled at the same time.

**Comment by klesh** [comment]
@antoniomuso I see, the deadlock problem is being worked on #7099, meanwhile you may set the Environment Variable to `PIPELINE_MAX_PARALLEL=1`  as a workaround.

**Comment by antoniomuso** [comment]
> @antoniomuso I see, the deadlock problem is being worked on #7099, meanwhile you may set the Environment Variable to `PIPELINE_MAX_PARALLEL=1` as a workaround.

Thank you; your suggestion works.

**Comment by klesh** [comment]
Closing this in favor of #7099
